project/Build.scala

```diff
@@ -211,65 +196,68 @@ object MimaBuild extends Build {
+
+  lazy val runTest = Def.task {
+    val cp = (fullClasspath in (reporterFunctionalTests, Compile)).value // the test classpath from the functionalTest project for the test
+    val proj = thisProjectRef.value // gives us the ProjectRef this task is defined in
+    val si = (scalaInstance in core).value // get a reference to the already loaded Scala classes so we get the advantage of a warm jvm
+    val v1 = (packageBin in v1Config).value // package the v1 sources and get the configuration used
+    val v2 = (packageBin in v2Config).value // same for v2
+    val scalaV = scalaVersion.value
+    val streams = Keys.streams.value
+    val urls = Attributed.data(cp).map(_.toURI.toURL).toArray
+    val loader = new java.net.URLClassLoader(urls, si.loader)
+
+    val testClass = loader.loadClass("com.typesafe.tools.mima.lib.CollectProblemsTest")
+    val testRunner = testClass.newInstance().asInstanceOf[{
+      def runTest(testClasspath: Array[String], testName: String, oldJarPath: String, newJarPath: String, oraclePath: String): Unit
+    }]
+
+    // Add the scala-library to the MiMa classpath used to run this test
+    val testClasspath = Attributed.data(cp).filter(_.getName endsWith "scala-library.jar").map(_.getAbsolutePath).toArray
+
+    val projectPath = proj.build.getPath + "reporter" + "/" + "functional-tests" + "/" + "src" + "/" + "test" + "/" + proj.project
+
+    val oraclePath = {
+      val p = projectPath + "/problems.txt"
+      val p212 = projectPath + "/problems-2.12.txt"
+      if(!(scalaV.startsWith("2.10.") || scalaV.startsWith("2.11.")) && new  java.io.File(p212).exists) p212
+      else p
+    }
+
+    try {
+      import scala.language.reflectiveCalls
+      testRunner.runTest(testClasspath, proj.project, v1.getAbsolutePath, v2.getAbsolutePath, oraclePath)
+      streams.log.info("Test '" + proj.project + "' succeeded.")
+    } catch {
+      case e: Exception =>  sys.error(e.toString)
+    }
+    ()
+  }
+
+  lazy val runAllTests = Def.taskDyn {
+    val s = state.value // this is how we access all defined projects from a task
+    val proj = thisProjectRef.value // gives us the ProjectRef this task is defined in
+    val _ = (test in Test).value // requires unit tests to run first
+
+    // gets all defined projects, dropping this project (core) so the task doesn't depend on itself
+    val structure = Project.structure(s)
+    val allProjects = structure.units(proj.build).defined.values filter (_.id != proj.project)
+
+    // get the fun-tests task in each project
+    val allTests = allProjects.toSeq flatMap { p => functionalTests in ProjectRef(proj.build, p.id) get structure.data }
+
+    // depend on all fun-tests
+    Def.task {
+      allTests.join.map(_ => ()).value
+    }
+  }
+
+  def project(id: String, base: File, settings: Seq[Def.Setting[_]] = Nil) =
+    Project(id, base, settings = settings) disablePlugins(BintrayPlugin)
```

Comment:
If you eliminate whitespace changes you'll see that that's def isn't new/my doing: https://github.com/typesafehub/migration-manager/pull/130/files?w=1

But sure, sounds much better than the status quo.